querygen/querygen.go

```diff
@@ -87,3 +87,25 @@ func GenClairScanCompletedQuery(index string) string {
 	)
 	return q
 }
+
+// GenReleaseCompletedQuery returns a Splunk query for generating Segment events
+// representing the Release resource success/failure state changes.
+func GenReleaseCompletedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=patch `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="appstudio.redhat.com" `+
+			`"objectRef.resource"="releases" `+
+			`"objectRef.subresource"="status" `+
+			`"responseObject.metadata.resourceVersion"="*" `+
+			`"responseObject.status.completionTime"="*" `+
+			`| spath responseObject.status.conditions{} `+
+			`| mvexpand responseObject.status.conditions{} `+
+			`| search responseObject.status.conditions{}.type="Released" AND `+
+			`(responseObject.status.conditions{}.reason="Succeeded" OR responseObject.status.conditions{}.reason="Failed") `+
+			`| eval event="Release completed ".mvindex('responseObject.status.conditions{}.reason', 0)`,
+		[]string{"namespace", "status_message", "status_reason"},
```

Comment:
Let's also include `"name"` in the list of output fields. Otherwise the resulting events all look practically the same.